### PR TITLE
add idiom 270 Test for quiet or signaling NaN

### PR DESCRIPTION
**Pull request by simonvonhackewitz**
Idiom 270 [Test for quiet or signaling NaN](https://www.programming-idioms.org/idiom/270/test-for-quiet-or-signaling-nan)

### CODE REVIEW

File: src/idiom270.fuzion [hunk 1]
```diff
@@ -0,0 +1 @@
+if f64.is_NaN r1 then "This is a NaN" else "This is a number"
```
Comment (fridis) [inline src/idiom270.fuzion:1]: This is not what the spec says: 
```
Given a floating point number r1 classify it as follows:
If it is a signaling NaN, print "This is a signaling NaN."
If it is a quiet NaN, print "This s a quiet NaN."
If it is not a NaN, print "This is a number."
```
Split code into several lignes.

File: src/idiom270ex.fz [hunk 2]
```diff
@@ -0,0 +1,6 @@
+ex270 is
+
+  numbers := [f64.NaN, f64.quiet_NaN, (0.0/0.0), 42.0, (1.0/0.0)]
+
+  for n in numbers do
+    if f64.is_NaN n then say "This is a NaN" else say "$n is a number"
```
Comment (fridis) [inline src/idiom270ex.fz:6]: use `r1` as variable here, not `n` such that the code can be the same as above.

**Review by fridis — CHANGES_REQUESTED** [review]

File: src/idiom270.fuzion (hunk 1 above)
Comment (simonvonhackewitz) [reply src/idiom270.fuzion:1]: That's right, but if I understood it correctly, Fuzion does not differentiate between quiet and signalling NaNs and treats all NaNs as quiet.

Would it be better to change the output for the NaN case to `"This is a quiet NaN"`?

Comment (michaellilltokiwa) [reply src/idiom270.fuzion:1]: @simonvonhackewitz You can create an issue for this in the main repo.

**Pull request merged**